libipl/libipl.C

```diff
@@ -359,3 +361,8 @@ void ipl_error_callback(bool status)
 		return;
 	g_ipl_error_callback_fn(status);
 }
+
+void ipl_set_apply_guards(bool apply)
```

Comment:
> OP1: since this initialization value for IPL, extend ipl_init function , add optional parameter . and default value can set as "true"
> OP2: Cleanest design would be int ipl_init(enum ipl_mode mode) -> int ipl_init(struct settings)

Agree, if we switched to  `int ipl_init(struct settings)` then we no need to add many `ipl_set_XXXX()` but, yes as you mentioned this prototype changes will impact all **libipl** user so before making changes I like to set design direction on this.

Reply:
I like the idea of passing a structure to `ipl_init()` for settings.  Only thing is, one should be able to pass NULL for default settings or leave a structure element not set to indicate use default value.  That means none of the settings can be boolean.

In this case, adding an api to ignore or apply guard records is fine.  However, your naming is horrible.  Since the default is to apply guard records, I would rename the api as `ipl_ignore_guard()`, that way there is no need to pass any arguments.

Reply:
Introduced the `ipl_settings` context.